satpy/readers/nc_slstr.py

```diff
@@ -112,21 +116,24 @@ def __init__(self, filename, filename_info, filetype_info):
 
     def get_dataset(self, key, info):
         """Load a dataset."""
-        if self.channel != key.name:
+
+        if self.channel not in key.name:
```

Comment:
What is it you want to cover here ?

Reply:
I could be wrong but this will do `'S2' in 'S2_n'` and do `'S2' in 'S2_o'`.

Reply:
Yes, that is correct. It ensures that we are reading from the correct file.